central/delegatedregistryconfig/service/service.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/stackrox/rox/pkg/grpc/authz/or"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/set"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		or.SensorOrAuthorizer(user.With(permissions.View(resources.Administration))): {
+			"/v1.DelegatedRegistryConfigService/GetConfig",
+		},
+		user.With(permissions.View(resources.Administration)): {
+			"/v1.DelegatedRegistryConfigService/GetClusters",
+		},
+		user.With(permissions.Modify(resources.Administration)): {
+			"/v1.DelegatedRegistryConfigService/PutConfig",
+		},
+	})
+)
+
+// Service provides the interface to modify the delegated registry config
+type Service interface {
+	pkgGRPC.APIService
+
+	AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error)
+
+	v1.DelegatedRegistryConfigServiceServer
+}
+
+// New returns a new Service instance using the given DataStore.
+func New(dataStore datastore.DataStore, clusterDataStore cluster.DataStore) Service {
+	return &serviceImpl{
+		dataStore:        dataStore,
+		clusterDataStore: clusterDataStore,
+	}
+}
+
+type serviceImpl struct {
+	v1.UnimplementedDelegatedRegistryConfigServiceServer
+
+	dataStore        datastore.DataStore
+	clusterDataStore cluster.DataStore
+}
+
+// RegisterServiceServer registers this service with the given gRPC Server.
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	v1.RegisterDelegatedRegistryConfigServiceServer(grpcServer, s)
+}
+
+// RegisterServiceHandler registers this service with the given gRPC Gateway endpoint.
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	return v1.RegisterDelegatedRegistryConfigServiceHandler(ctx, mux, conn)
+}
+
+// AuthFuncOverride specifies the auth criteria for this API.
+func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, authorizer.Authorized(ctx, fullMethodName)
+}
+
+// GetConfig returns Central's delegated registry config
+func (s *serviceImpl) GetConfig(ctx context.Context, _ *v1.Empty) (*v1.DelegatedRegistryConfig, error) {
+	config, exists, err := s.dataStore.GetConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving config %w", err)
+	}
+
+	if !exists {
+		return &v1.DelegatedRegistryConfig{}, nil
+	}
+
+	return convert.StorageToAPI(config), nil
+}
+
+// GetClusters returns the list of clusters (id + name) that are eligible for delegating scanning
+// requests (ie: only clusters with scanners that understand the delegated registry config)
+func (s *serviceImpl) GetClusters(ctx context.Context, _ *v1.Empty) (*v1.DelegatedRegistryClustersResponse, error) {
+	clusters, err := s.getClusters(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving clusters %w", err)
```

Comment:
I was going to to mention `status.Error` here, but I see it is used below. Curious, why not here, too?

Reply:
returning an error defaults to a `500` which is whats desired in those cases, was only using it when needed to change the status code to a `400` , was modeling this off of the config & cluster services which split usage as well